src/System.CommandLine/src/System/CommandLine/ArgumentSyntax_Definers.cs

```diff
@@ -53,30 +52,73 @@ public Argument<int> DefineOption(string name, int defaultValue)
             return DefineOption(name, defaultValue, s_int32Parser);
         }
 
-        public Argument<T> DefineOption<T>(string name, ref T value, Func<string, T> valueConverter, string help)
+        public Argument<T> DefineOption<T>(
+            string name,
+            Action<T> assign,
```

Comment:
Sure, I can change the name.

However, I am not sure I follow what you mean by the two methods. I understand option 1, something that takes the delegate and always runs it. But what would be the params to option 2? We need one for the assignment, so that we can assign to properties, for instance, and one to do the check. The goal here is to not have to do a bunch of if () variable = value kind of code in the parser.